src/Altinn.Broker.Core/Helpers/ManifestDownloadStream.cs

```diff
@@ -0,0 +1,262 @@
+        }
+
+        return Task.FromCanceled<int>(cancellationToken);
+    }
+
+    public override void Flush()
+    {
+    }
+
+    public override Task FlushAsync(CancellationToken cancellationToken)
+    {
+        return Task.CompletedTask;
+    }
+
+    public override void SetLength(long value)
+    {
+        throw new NotSupportedException();
+    }
+
+    public override void Write(byte[] buffer, int offset, int count)
+    {
+        throw new NotSupportedException();
+    }
+
+    private static void ValidateReadArrayArguments(byte[] buffer, int offset, int count)
+    {
+        if (buffer == null)
+        {
+            throw new ArgumentNullException("buffer");
+        }
+
+        if (offset < 0)
+        {
+            throw new ArgumentOutOfRangeException("offset");
+        }
+
+        if (count < 0 || buffer.Length - offset < count)
+        {
+            throw new ArgumentOutOfRangeException("count");
+        }
+    }
+
+    private void ValidateNotClosed()
+    {
+        if (!_isOpen)
+        {
+            throw new ObjectDisposedException(null, "Cannot access a closed Stream");
+        }
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        try
+        {
+            if (disposing)
+            {
+                _isOpen = false;
+                _content = default(ReadOnlyMemory<byte>);
+            }
+        }
+        finally
+        {
+            base.Dispose(disposing);
+        }
+    }
+
+
+    public async Task AddManifestFile(FileTransferEntity fileTransferEntity)
+    {
+        ValidateNotClosed();
+
+        if (!IsZipFile())
+        {
+            throw new InvalidOperationException("The stream must contain a valid ZIP archive");
+        }
+
+        // Create a memory stream to hold our modified ZIP content
+        using var modifiedZipStream = new MemoryStream();
+
+        // Copy current content to the working memory stream
+        Position = 0;
+        await CopyToAsync(modifiedZipStream);
+        modifiedZipStream.Position = 0;
+
+        using (var archive = new ZipArchive(modifiedZipStream, ZipArchiveMode.Update, true))
+        {
+            // Remove existing manifest files
+            var manifestEntry = archive.GetEntry("Manifest.xml");
+            manifestEntry?.Delete();
+
+            var recipientsEntry = archive.GetEntry("Recipients.xml");
+            recipientsEntry?.Delete();
+
+            // Create new manifest entry
+            var newManifestEntry = archive.CreateEntry("Manifest.xml");
+            using (var manifestStream = newManifestEntry.Open())
+            {
+                var manifest = fileTransferEntity.CreateManifest();
+                WriteSerializeManifestToZip(manifest, manifestStream);
+            }
+        }
+
+        // Update the content of our DownloadStream with the modified ZIP
+        _content = new ReadOnlyMemory<byte>(modifiedZipStream.ToArray());
+        _position = 0;
+    }
```

Comment:
No can do as the base stream is a ReadOnlyMemoryStream based on a download stream. Will evaluate if it can be done when download stream is re-written in the "Support bigger files"-PR.